CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+Thanks for your interest in the Terraform Provider for Juju -- contributions like yours make good projects
+great!
+
+Whether it is code or docs, there are two basic ways to contribute: by opening
+an issue or by creating a PR. This document gives detailed information about
+both.
+
+> Note: If at any point you get stuck, come chat with us on
+[Matrix](https://matrix.to/#/#terraform-provider-juju:ubuntu.com).
+
+## Open an issue
+
+You will need a GitHub account ([sign up](https://github.com/signup)).
+
+### Open an issue for docs
+
+To open an issue for a specific doc, find [the published doc](https://canonical-terraform-provider-juju.readthedocs-hosted.com/en/latest/tutorial/), then use the **Give feedback** button.
+
+To open an issue for docs in general, do the same for the homepage of the docs
+or go to https://github.com/juju/terraform-provider-juju/issues, click on **New issue** (top right corner
+of the page), select “Blank issue”, then fill out the issue template and
+submit the issue.
+
+### Open an issue for code
+
+Go to https://github.com/juju/terraform-provider-juju/issues  click on **New issue** (top right
+corner of the page), select whatever is appropriate, then fill out the issue
+template and submit the issue.
+
+> Note: For feature requests please use
+https://matrix.to/#/#terraform-provider-juju:ubuntu.com
+
+## Make your first contribution
+
+You will need a GitHub account ([sign up](https://github.com/signup) and [add
+your public SSH key](https://github.com/settings/ssh)) and `git` ([get
+started](https://git-scm.com/book/en/v2/Getting-Started-What-is-Git%3F)).
+
+Then:
+
+1. [Sign the Canonical Contributor Licence Agreement
+   (CLA)](https://ubuntu.com/legal/contributors).
+
+2. Configure your `git` so your commits are signed:
+
+```
+git config --global user.name "A. Hacker"
+git config --global user.email "a.hacker@example.com"
+git config --global commit.gpgsign true
+```
+
+> See more: [GitHub | Authentication > Signing commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+
+3. Fork juju/terraform-provider-juju. This will create `https://github.com/<user>/terraform-provider-juju`.
+
+4. Clone your fork locally.
+
+```
+git clone git@github.com:<user>/terraform-provider-juju.git
+```
+
+5. Add a new remote with the name `upstream` and set it to point to the upstream
+`juju` repo.
+
+```
+git remote add upstream git@github.com:juju/terraform-provider-juju.git
+```
+
+6. Set your local branches to track the `upstream` remote (not your fork). E.g.,
+
+```
+git checkout main
+git branch --set-upstream-to=upstream/main
+```
+
+7. Sync your local branches with the upstream, then check out the branch you
+want to contribute to and create a feature branch based on it.
+
+```
+git fetch upstream
+git checkout main
+git pull
+git checkout -b main-new-stuff # your feature branch
+```
+
+8. Make the desired changes. Test changes locally.
```

Comment:
Something of a todo is that we should have a contributing.md file and direct users there to understand how to run tests, etc locally.